### PR TITLE
Using modules and smaller Docker image

### PR DESCRIPTION
**Pull request by pablogrs**
Hi,

Thanks for the boiler plate app, much simplified the work of creating a GH app.
Just took the opportunity to use Go 1.17, use modules and reduce the Docker image size

### CODE REVIEW

**Comment by pablogrs** [comment]
@sharkySharks I hope you can review this.
I think this variable is not really being used, just assigned
image

File: Dockerfile [hunk 1]
```diff
@@ -12,15 +12,19 @@ WORKDIR $GOPATH/src/github.com/sharkySharks/go-github-app-boilerplate/
...
 ENV env=${env}
 
-CMD ["/go/bin/main"]
+FROM alpine:3.12
 
+# Copy our static executable.
+COPY --from=builder /go/bin/main /bin/main
+COPY --from=builder /secrets.yaml /secrets.yaml
+
+CMD ["/bin/main"]
```
Comment (sharkySharks) [inline Dockerfile:30]: let's add a new line

File: main/go.mod [hunk 2]
```diff
@@ -0,0 +1,21 @@
+module github.hpe.com/pablo-gon-sanchez/go-github-app-boilerplate/main
```
Comment (sharkySharks) [inline main/go.mod:1]: is there a way we can genericize this module name so it's not specific to local directory structure? If not, then I think for the purposes of this repo the module should be
```
module github.com/sharkysharks/go-github-app-boilerplate
```
you don't need to have `/main` on the end, the modules will work the same inside the main directory.

**Review by sharkySharks — COMMENTED** [review]
thanks so much for the go modules update! I had made these changes in one of my applications but haven't back ported the changes here yet. I also have a serverless version I just started using too so maybe I'll make a serverless branch on here as well. 

By moving to go modules we can actually delete the `vendor/` directory and `Gopkg.lock` and `Gopkg.toml`, since those are used with `dep` and not `mod`.

If you could take a minute to update the readme to reflect the changes here that would be great too, like no need to have a specific directory structure with go mod, `dep` is no longer the tool to build the app, etc.

**Comment by sharkySharks** [comment]
@pablogrs the purpose of initializing `ghClient` is so that you can use it after you receive an action, so you can then make calls back to github with whatever actions you want to take. So yes, in this boilerplate code it is not being used anywhere after initialization, but in the apps I have made with this I will then use it to add labels to pull requests or leave comments or any other github api call. you can think of that you might need it for.

**Comment by pablogrs** [comment]
Awesome, I'll make use of ghClient for sure then. thanks for the reply and yes I'll update the Readme.

File: Dockerfile (hunk 1 above)
Comment (pablogrs) [reply Dockerfile]: I didn't get it so I added a comment instead :smile_cat:

File: main/go.mod (hunk 2 above)
Comment (pablogrs) [reply main/go.mod]: I changed it to not use the main, only thing is the executable file has changed its name, check if you like that please?

**Comment by pablogrs** [comment]
I don't if it is correct but I have a branch where I divide the packages middleware and config 
https://github.com/pablogrs/go-github-app-boilerplate/tree/divide-in-packages

**Comment by sharkySharks** [comment]
Hey @pablogrs I havent had a chance to look at it yet but thanks for poking me. I'll take a look and get back to you asap.

File: main/go.mod [hunk 3]
```diff
@@ -0,0 +1,21 @@
+module github.hpe.com/sharkysharks/go-github-app-boilerplate
```
Comment (sharkySharks) [inline main/go.mod:1]: i think this just needs to be `github.com/sharkysharks`. minus the `hpe`

**Review by sharkySharks — COMMENTED** [review]
you could move middleware and config into their own packages like you have in your fork, i have organized it both ways, just depends on what other packages you may be creating. I have them in separate directories when I made these serverless functions and I ended up making two serverless functions which need to be in separate directories and then made a github client directory so I could share that package across both lambdas. so just depends on what you are working with. keeping them together is simple until you need to separate them.

thanks for doing all the work to get go modules working in here, much appreciated. Just a small change to the module and I'll sign off!

File: main/go.mod (hunk 3 above)
Comment (pablogrs) [reply main/go.mod:1]: oops!

**Review by sharkySharks — APPROVED** [review]
Awesome 🎉

**Pull request merged**